community/cypher/frontend-3.0/src/main/scala/org/neo4j/cypher/internal/frontend/v3_0/ast/functions/ToFloat.scala

```diff
@@ -28,6 +28,7 @@ case object ToFloat extends Function with SimpleTypedFunction {
   val signatures = Vector(
     Signature(argumentTypes = Vector(CTFloat), outputType = CTFloat),
     Signature(argumentTypes = Vector(CTInteger), outputType = CTFloat),
-    Signature(argumentTypes = Vector(CTString), outputType = CTFloat)
+    Signature(argumentTypes = Vector(CTString), outputType = CTFloat),
+    Signature(argumentTypes = Vector(CTNumber), outputType = CTFloat)
```

Comment:
If you add CTNumber, there's no need to keep CTInteger and CTFloat as separate signatures.